### PR TITLE
Adding aten::is_floating_point evaluator

### PR DESCRIPTION
**Pull request by narendasan**
# Description

This adds support for the `aten::is_floating_point` operator. Also adds a `aten::Int` evaluator that needs to be discussed

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project (You can use the linters)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas and hacks
- [X] I have made corresponding changes to the documentation
- [X] I have added tests to verify my fix or my feature
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/evaluators/aten.cpp [hunk 1]
```diff
@@ -464,7 +464,38 @@ auto aten_registrations TRTORCH_UNUSED =
...
+                      } else {
+                        auto tensor = tensor_var.unwrapToTensor();
+                        auto t = tensor.scalar_type();
+                        return at::isFloatingType(t);
+                      }
+                    },
+                    EvalOptions().validSchemas({
+                        "aten::is_floating_point(Tensor self) -> (bool)",
+                    })})
+        .evaluator({c10::Symbol::fromQualString("aten::Int"),
```
Comment (narendasan) [inline core/conversion/evaluators/aten.cpp:484]: Is there a better way to handle this? Type casts dont have meaning in TensorRT. Are what are the possible use cases of this op and how do we handle them?

File: core/conversion/evaluators/aten.cpp [hunk 2]
```diff
@@ -464,7 +464,38 @@ auto aten_registrations TRTORCH_UNUSED =
...
+                    [](const torch::jit::Node* n, kwargs& args) -> c10::optional<torch::jit::IValue> {
+                      auto tensor_var = args.at(n->input(0));
+                      if (tensor_var.isITensor()) {
+                        TRTORCH_ASSERT(!tensor_var.isITensor(), "Unable evaluate ITensor to int");
```
Comment (peri044) [inline core/conversion/evaluators/aten.cpp:488]: "Unable to evaluate ITensor to int in aten::int evaluator" maybe ?

File: core/conversion/evaluators/aten.cpp (hunk 1 above)
Comment (peri044) [reply core/conversion/evaluators/aten.cpp:484]: Tagging @inocsin for viz. I have similar question as well. Can anyone let me know the usage of this op ? 
In line 487, there is a check if node input is an ITensor or not. ITensor() true case happens when  node input is in ctx->value_tensor_map right ? What is the layer preceeding this layer usually ? Can you add test case if there are no blockers ?

**Review by peri044 — CHANGES_REQUESTED** [review]

File: core/conversion/evaluators/aten.cpp (hunk 2 above)
Comment (inocsin) [reply core/conversion/evaluators/aten.cpp:488]: Yeah, do you have any idea how to get the ITensor value in the evalutor?

File: core/conversion/evaluators/aten.cpp (hunk 1 above)
Comment (inocsin) [reply core/conversion/evaluators/aten.cpp:484]:
```
  %2270 : bool = aten::is_floating_point(%input174.1) # <string>:3:6
  %2271 : bool = prim::If(%2270) # <string>:3:6
```
It was used as in If-Else

Comment (inocsin) [reply core/conversion/evaluators/aten.cpp:484]: aten::Int, aten::is_floating_point test case is here https://github.com/NVIDIA/TRTorch/pull/296 , the evaluator test framework was not ready at that time, so I submit anothor commit with test case.

File: core/conversion/evaluators/aten.cpp (hunk 2 above)
Comment (narendasan) [reply core/conversion/evaluators/aten.cpp:488]: ITensors have no definite value at compile time. If an ITensor hits an evaluator compilation is going to fail.

**Comment by narendasan** [comment]
@peri044 is looking at casting so we might need to revisit this

**Pull request merged**